web3.js/src/transaction.js

```diff
@@ -141,6 +141,31 @@ export class Transaction {
     return null;
   }
 
+  /**
+   * The transaction fee payer (first signer)
+   */
+  get feePayer(): PublicKey | null {
+    if (this.signatures.length > 0) {
+      return this.signatures[0].publicKey;
+    }
+    return null;
+  }
+
+  /**
+   * Set the transaction fee payer (first signer)
+   */
+  set feePayer(feePayer: PublicKey) {
+    if (
+      this.signatures.length === 0 ||
+      !this.signatures[0].publicKey.equals(feePayer)
```

Comment:
We'd need to clear any existing signatures in this case.

It also pushes superfluous `null` sigs into the list in the case where this function is called multiple times with different, unreferenced `publicKey`s

Reply:
Good call, will fix

Reply:
@t-nelson I ended up dropping `set feePayer`